### ISSUE TITLE
Self-adjoint eigen decomposition was not successful. The input might not be valid.

### ISSUE THREAD

**Issue by yabmtm** [markovmodel/deeptime#33]
I'm trying to train a VAMPnet on one of my MD protein-ligand datasets. I've tried using input as both pair-wise distance features and cartesian coordinates of alpha carbons from PBC-corrected trajectories. I'm currently using Tensorflow 1.8 and the hyperparameters:

```
tau = 10
batch_size = 1000
training_percentage = 75
network_depth = 6
layer_width = 16
learning_rate = 5e-4
output_size = 8
nb_epoch = 10
epsilon = 1e-5
```

I've also attached a notebook that I've edited a bit from the alanine_dipepetide_multiple_files notebook in the examples directory.
I've been able to build traditional MSMs and MEMMs with the data, so it should be valid.
[VAMPnets.zip](https://github.com/markovmodel/deeptime/files/3795752/VAMPnets.zip)

**Comment by amardt** [comment]
Hey,

I did not have the time yet to check your notebook. But does this happen directly at the beginning of your training? Can you try it with a larger batchsize? This sounds like a numerical problem. You could also first test it with less layers. 
Let me know, if one of these things already help. Otherwise I will have a look at your notebook at a later point.

Best
Andreas

**Comment by yabmtm** [comment]
Thanks for your quick response. I increased the batch size to 10000 and everything seems to work fine. I've tested depths of 2,4,6,8, and increased width up to 64.  I'm going to close this for now as that seems to have been the problem. Thanks again!

**Comment by jameschapman19** [comment]
Thought I'd add here as I've come across this problem when using a CCA loss function and it might be helpful to someone to see here:

If your batch size leads to a batch of size 1 (perhaps at the end of an epoch), then the Self-adjoint eigen decomposition fails. I've ended up using hacky workarounds like checking the batch size and adjusting until N%batch_size =/= 1.
